AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/Support/Model+Compare.swift

```diff
@@ -0,0 +1,152 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+// swiftlint:disable cyclomatic_complexity
+extension ModelSchema {
+
+    private static let serviceUpdatedFields: Set = ["updatedAt", "createdAt"]
+
+    /// Compare two `Model` based on a given `ModelSchema`
+    /// Returns true if equal, false otherwise
+    /// Currently, schemas where system timestamps fields (createdAt, updatedAt)
+    /// are changed with `timestamps` attribute to model directive and also added
+    /// to model schema are not covered. This will cause `isEqual` to fail in
+    /// such cases.
```

Comment:
```suggestion
    /// Currently, schemas where system timestamps fields (createdAt, updatedAt)
    /// are renamed using with `@model`'s `timestamps` attribute and explicitly
    /// added to the input schema are not supported by this check since they are
    /// marked as "read-only" fields and will fail the check when the service generates
    /// and returns the value of `createdAt` or `updatedAt`.
```